### ISSUE TITLE
altera-id-dependent-backward-branch false positive

### ISSUE THREAD

**Issue by mumbleskates** [llvm/llvm-project#53777]
The following (relatively) minimal program seems to trigger a false positive for the `altera-id-dependent-backward-branch` inspection:

```c++
#include <initializer_list>

int main() {
  for (int chunk_size : {1, 2, 3}) {
    for (int i = 0; i < 500; i += chunk_size) {
      static_cast<void>(i);
    }
  }
}
```
"Clang-Tidy: Backward branch (for loop) is ID-dependent due to variable reference to 'i' and may cause performance degradation"

**Comment by LegalizeAdulthood** [comment]
Is a quick fix to simply disable this check when not compiling OpenCL code?

**Comment by vbvictor** [comment]
Probably related to https://github.com/llvm/llvm-project/issues/52790
